### PR TITLE
test: convert e2e test cases from old to new framework (part 4)

### PR DESCRIPTION
**Pull request by chewong**
**Reason for Change**:

Follow-up PR for #662.

Converted the following test cases from old to new framework:
`should pass the identity validation even when MIC leader is changed`
`should pass liveness probe test`
`should be backward compatible with old and new version of MIC and NMI`

**Issue Fixed**:

ref: #586, #657

**Notes for Reviewers**:

### CODE REVIEW

File: test/e2e_new/framework/azureidentity/azureidentity_helpers.go [hunk 1]
```diff
@@ -73,6 +79,63 @@ func Create(input CreateInput) *aadpodv1.AzureIdentity {
 	return azureIdentity
 }
 
+// CreateOldInput creates an old AzureIdentity resource.
```
Comment (aramase) [inline test/e2e_new/framework/azureidentity/azureidentity_helpers.go:82]: Can we add more info in this comment about what "old AzureIdentity" means?

File: test/e2e_new/framework/azureidentitybinding/azureidentitybinding_helpers.go [hunk 2]
```diff
@@ -60,6 +66,59 @@ func Create(input CreateInput) *aadpodv1.AzureIdentityBinding {
 	return azureIdentityBinding
 }
 
+// CreateOldInput creates an old AzureIdentity resource.
```
Comment (aramase) [inline test/e2e_new/framework/azureidentitybinding/azureidentitybinding_helpers.go:69]: s/AzureIdentity/AzureIdentityBinding

File: test/e2e_new/framework/azureidentitybinding/azureidentitybinding_helpers.go [hunk 3]
```diff
@@ -60,6 +66,59 @@ func Create(input CreateInput) *aadpodv1.AzureIdentityBinding {
...
+func CreateOld(input CreateInput) string {
+	type azureIdentityBindingOld struct {
+		APIVersion string `json:"apiVersion"`
+		Kind       string `json:"kind"`
+		*aadpodv1.AzureIdentityBinding
+	}
+
+	Expect(input.Name).NotTo(BeEmpty(), "input.Name is required for AzureIdentity.Create")
```
Comment (aramase) [inline test/e2e_new/framework/azureidentitybinding/azureidentitybinding_helpers.go:77]: s/AzureIdentity/AzureIdentityBinding for all the checks

File: test/e2e_new/framework/helm/helm_helpers.go [hunk 4]
```diff
@@ -64,8 +64,26 @@ func Uninstall() {
 	})
 }
 
-func Upgrade() {
-	// TODO
+func Upgrade(config *framework.Config) {
+	cwd, err := os.Getwd()
+	Expect(err).To(BeNil())
+
+	// Change current working dirrectory to repo root
```
Comment (aramase) [inline test/e2e_new/framework/helm/helm_helpers.go:71]: s/dirrectory/directory

File: test/e2e_new/backward_compat_test.go [hunk 5]
```diff
@@ -3,11 +3,117 @@
...
+			"--ignore-not-found",
+			"cm",
+			"aad-pod-identity-config",
+		})
+		Expect(err).To(BeNil())
+
+		configOldVersion := config.DeepCopy()
+		configOldVersion.Registry = "mcr.microsoft.com/k8s/aad-pod-identity"
+		configOldVersion.MICVersion = "1.5"
+		configOldVersion.NMIVersion = "1.6.0"
```
[the remark this answers is not in the document]
Comment (chewong) [reply test/e2e_new/backward_compat_test.go:66]: The reason is that the current helm chart is backward incompatible with older version of MIC and NMI (for example, NMI v1.4 does not support `--operation-mode` flag and the older version does not `ENTRYPOINT` in the Dockerfile, meaning we have to supply `command` in the YAML deployment). I will modify the helm chart accordingly to it compatible with older versions.

[the remark this answers is not in the document]
Comment (aramase) [reply test/e2e_new/backward_compat_test.go]: That makes sense. Let's make the backward compat test for 1.5 to 1.6 as 1.4 release is almost a year old now. Then you don't need to add the `ENTRYPOINT` to helm chart.

File: test/e2e_new/framework/azureidentity/azureidentity_helpers.go [hunk 6]
```diff
@@ -73,6 +79,65 @@ func Create(input CreateInput) *aadpodv1.AzureIdentity {
...
+// The JSON fields of old AzureIdentity have their first letter capitalized, which we do not support fior v1.6.0 and onward.
```
Comment (aramase) [inline test/e2e_new/framework/azureidentity/azureidentity_helpers.go:83]: nit: s/fior/for

File: test/e2e_new/framework/azureidentitybinding/azureidentitybinding_helpers.go [hunk 7]
```diff
@@ -60,6 +66,61 @@ func Create(input CreateInput) *aadpodv1.AzureIdentityBinding {
 	return azureIdentityBinding
 }
 
+// CreateOldInput creates an old AzureIdentityBinding resource.
+// The JSON fields of old AzureIdentityBinding have their first letter capitalized, which we do not support fior v1.6.0 and onward.
```
Comment (aramase) [inline test/e2e_new/framework/azureidentitybinding/azureidentitybinding_helpers.go:70]: nit: s/fior/for

**Pull request merged**